### PR TITLE
Fixes #11576

### LINKED ISSUE #11576: "You're on fire."
(WEB REPORT BY: repertor REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> 8b80edba9d7c2adce0bd948ccd5d8ef840172871
> 
> General description of the issue
> as a diona, getting lit on fire by a pie tin will give you flames that don't go out, causing you to glow and look awesome.
> 
> What you expected to happen
> IT BURNS OH GOD IT BURNS
> 
> What actually happened
> you look pretty cool instead.
> 
> Steps to reproduce if possible
> be diona. Make pie tin fire. Step into fire. Flame does not go out.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #11576
also rejigs a istype(src) because that's awful

### CODE REVIEW

File: code/modules/mob/living/carbon/human/life/handle_fire.dm [hunk 1]
```diff
@@ -3,7 +3,8 @@
 /mob/living/carbon/human/handle_fire()
 	if(..())
 		return
+	if(mind &&.mind.vampire && stat == DEAD)
```
Comment (Intigracy) [inline code/modules/mob/living/carbon/human/life/handle_fire.dm]: You forgot something

**Review by Intigracy — DISMISSED** [review]

File: code/modules/mob/living/carbon/human/life/handle_fire.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_fire.dm]: This is the exact sanity from below

Comment (Intigracy) [reply code/modules/mob/living/carbon/human/life/handle_fire.dm]: Remove the .

Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_fire.dm]: whoops

**Pull request merged**